frontend/src/features/env/EnvService.ts

```diff
@@ -1,7 +1,83 @@
-import { ClientEnvVars } from '~shared/types/core'
+import {
+  ErrorDto,
+  PublicFormViewDto,
+  switchEnvFeedbackFormBodyDto,
+} from '~shared/types'
+import { ClientEnvVars, SuccessMessageDto } from '~shared/types/core'
 
+import { transformAllIsoStringsToDate } from '~utils/date'
 import { ApiService } from '~services/ApiService'
 
+import { PUBLIC_FORMS_ENDPOINT } from '~features/public-form/PublicFormService'
+
 export const getClientEnvVars = async (): Promise<ClientEnvVars> => {
   return ApiService.get<ClientEnvVars>('/client/env').then(({ data }) => data)
 }
+
+// TODO #4279: Remove after React rollout is complete
+const createFeedbackResponsesArray = (
+  formInputs: switchEnvFeedbackFormBodyDto,
+  feedbackForm: PublicFormViewDto,
+) => {
+  const responses = []
+  for (const [key, value] of Object.entries(formInputs)) {
+    const fieldIndex = key === 'url' ? 0 : key === 'feedback' ? 1 : 2
+    const { _id, fieldType } = feedbackForm?.form.form_fields[fieldIndex] ?? {}
+    const entry = {
+      _id,
+      question: key,
+      answer: value ?? '',
+      fieldType,
+    }
+    responses.push(entry)
```

Comment:
Re-reading the code, the structure of the loop reads a little "strange". The loop heading is generic (using Object.entries()`), but the loop body shows very explicit expectations of having just 3 entries, 2 of which with known names.

Clearly we're working on mapping the 2 structure `formInputs` and `feedbackForm`, with their shapes being well defined. Because of that, the extraction code could make that mapping explicit instead, to show what we expect to get at the end, maybe something like this:

```suggestion
  // [key_in_inputs:string, corresponding_index_in_form:number]
  return [
    ['url', 0],
    ['feedback', 1],
    ['email', '2']
  ]
    .map(([inputKey, formFieldIndex]) => {
      const { _id, fieldType } = feedbackForm.form.form_fields[formFieldIndex]
      const answer = formInputs[inputKey] ?? ''
  
      return {
        _id,
        question: inputKey,
        answer,
        fieldType,
      }
    })
```

Some note on the above:
1. untested (am not sure if typescript will be happy with it 😅)
2. It gives explicit order of the resultant response array
3. it _looks_ like `formFieldIndex` are not needed in the map definition because the index values match the array indexes anyway, but I think it's worth having them stated anyway, because the indexes represent different "things". For example, this would work as a map definition, and show the indexes are not necessarily synced:
```javascript
  return [
    ['url', 0],
    ['email', '2']
    ['feedback', 1],
  ]
    .map( /* ... */ )
```

Reply:
thanks for the extensive comment and explanation Tim! It does make it clearer to make the mapping explicit. I've taken your suggestion and added type definitions [here](https://github.com/opengovsg/FormSG/pull/4352/commits/cad0025747635dea10b5d6bedf5dfbe0dca7dfa8)